pkg/utils/secrets/rsa_private_key.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secrets
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+
+	"github.com/gardener/gardener/pkg/utils"
+	"golang.org/x/crypto/ssh"
+)
+
+// RSASecret containing information about the number of bits which should be used for the to-be-created RSA private key.
+type RSASecret struct {
```

Comment:
If this is just the configuration used to generate an actual `secret.RSA`, why isn't it called `secret.RSAConfig`?

Reply:
I would go for RSASecretConfig.